stdlib/pqueue.mli

```diff
@@ -0,0 +1,208 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Jean-Christophe Filliâtre                        *)
+(*                                                                        *)
+(*   Copyright 2023 CNRS                                                  *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Priority queues.
+
+    The {!Pqueue} module implements a data structure of priority queues,
+    given a totally ordered type for priorities. This is a mutable
+    data structure. Smaller values are considered to have a higher
+    priority (i.e., these are min-priority queues).
+
+    The implementation uses a heap stored in a dynamic array, and is
+    therefore reasonably efficient: accessing the minimum element
+    takes constant time, and insertion and removal take time
+    logarithmic in the size of the priority queue. Note that
+    [of_array] runs in linear time (and thus must be preferred to
+    repeated insertions with [add]).
+
+    It is fine to have several elements with the same priority.
+    It is guaranteed that the element returned by [min_elt] (or
+    [min_elt_opt]) is the one that is removed from the priority queue
+    by [pop_min] (or [pop_min_opt] or [remove_min]). This is important
+    in many algorithms, (e.g. when peeking at several priority queues
+    and then selecting one to remove from).
+
+    @since 5.2
+*)
+
+module type OrderedType =
+  sig
+    type t
+      (** The type of priorities. *)
+
+    val compare : t -> t -> int
+      (** A total ordering function to compare priorities.
+
+          This is a two-argument function [f] such that [f p1 p2] is
+          zero if the priorities [p1] and [p2] are equal,
+          [f p1 p2] is strictly negative if [p1] has higher priority than
+          [p2], and [f p1 p2] is strictly positive if [p1]
+          has lower priority than [p2].
```

Comment:
I found this paragraph a bit confusing for non-experts. "higher priority" would be understood in informal language as "the thing that must be done first", and you mean exactly the opposite here. Is there a way to reformulate this to avoid this confusion? Or maybe we could point it out explicitly somewhere in the documentation?

Reply:
Hi @gasche,
I tried to address these concerns in 3c53d82c, with two different signatures `Min` and `Max` and two different functors `MakeMin` and `MakeMax`. Comments are welcome.